contracts/oracle/VoltSystemOracle.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {Timed} from "./../utils/Timed.sol";
+import {Constants} from "./../Constants.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 1 year period into the VOLT price
+/// after the oracle start time.
+/// Interest can compound annually.
+/// @author Elliot Friedman
+contract VoltSystemOracle is Timed, Initializable, IVoltSystemOracle {
+    /// ---------- Mutable Price Variable ----------
+
+    /// @notice acts as an accumulator for checkpointing interest earned in previous periods
+    uint256 public override oraclePrice;
+
+    /// ---------- Immutable Variables ----------
+
+    /// @notice current amount that oracle price is inflating by yearly in basis points
+    uint256 public immutable annualChangeRateBasisPoints;
+
+    /// @notice the time frame over which all changes in the APR are applied
+    /// one year was chosen because this is a temporary oracle
+    uint256 public constant override TIMEFRAME = 365 days;
+
+    /// @notice start time at which point interest will start accruing, and the
+    /// point in time at which the current ScalingPriceOracle price will be
+    /// snapshotted and saved
+    uint256 public immutable oracleStartTime;
+
+    /// @param _annualChangeRateBasisPoints yearly change rate in the Volt price
+    /// @param _oracleStartTime start time at which oracle starts interpolating prices
+    /// @param _oraclePrice starting oracle price
+    constructor(
+        uint256 _annualChangeRateBasisPoints,
+        uint256 _oracleStartTime,
+        uint256 _oraclePrice
```

Comment:
How can this `_oraclePrice` be known if it's in the future?

Reply:
because the current scaling price oracle has a deterministic price at the end of the month. so we know the price will be a specific number after the interpolation has completed.